ios/ApptentiveModule.swift

```diff
@@ -94,8 +94,8 @@ import ApptentiveKit
   // Add person custom data based on key string and value of type bool, number, or string
   @objc(addCustomPersonData:value:resolver:rejecter:)
   func addCustomPersonData(_ key: String, value: Any, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
-    if let value_bool = value as? Bool {
-      Apptentive.shared.personCustomData[key] = value_bool
+    if let value_number = value as? NSNumber, CFGetTypeID(value_number) == CFBooleanGetTypeID() {
```

Comment:
The problem is `Any` cannot be type casted to `Bool` directly?

Reply:
The issue is that the type-casting to Bool in the old way is "greedy": it will translate "false" and "true" to bool as well as 0 and 1. And the backend and targeting treat 0 and 1 differently from false and true. 
So we need to dive deep into the guts of CoreFoundation to actually figure out the underlying type that the JS layer handed off to the Objective-C layer.